### ISSUE TITLE
[BUG]: GPU is not actually utilized

### ISSUE THREAD

**Issue by rtj563etn36h4h** [serengil/deepface#1328]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.93

### Python version

3.10

### Operating System

Ubuntu WSL2

### Dependencies

absl-py==2.1.0
astunparse==1.6.3
beautifulsoup4==4.12.3
blinker==1.8.2
certifi==2024.8.30
charset-normalizer==3.3.2
click==8.1.7
deepface==0.0.93
filelock==3.15.4
fire==0.6.0
Flask==3.0.3
Flask-Cors==5.0.0
flatbuffers==24.3.25
gast==0.6.0
gdown==5.2.0
google-pasta==0.2.0
grpcio==1.66.1
gunicorn==23.0.0
h5py==3.11.0
idna==3.8
itsdangerous==2.2.0
Jinja2==3.1.4
keras==3.5.0
libclang==18.1.1
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==2.1.5
mdurl==0.1.2
ml-dtypes==0.4.0
mtcnn==0.1.1
namex==0.0.8
numpy==1.26.4
nvidia-cublas-cu12==12.3.4.1
nvidia-cuda-cupti-cu12==12.3.101
nvidia-cuda-nvcc-cu12==12.3.107
nvidia-cuda-nvrtc-cu12==12.3.107
nvidia-cuda-runtime-cu12==12.3.101
nvidia-cudnn-cu12==8.9.7.29
nvidia-cufft-cu12==11.0.12.1
nvidia-curand-cu12==10.3.4.107
nvidia-cusolver-cu12==11.5.4.101
nvidia-cusparse-cu12==12.2.0.103
nvidia-nccl-cu12==2.19.3
nvidia-nvjitlink-cu12==12.3.101
opencv-python==4.10.0.84
opt-einsum==3.3.0
optree==0.12.1
packaging==24.1
pandas==2.2.2
pillow==10.4.0
protobuf==4.25.4
Pygments==2.18.0
PySocks==1.7.1
python-dateutil==2.9.0.post0
pytz==2024.1
requests==2.32.3
retina-face==0.0.17
rich==13.8.0
six==1.16.0
soupsieve==2.6
tensorboard==2.17.1
tensorboard-data-server==0.7.2
tensorflow==2.17.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.4.0
tf_keras==2.17.0
tqdm==4.66.5
typing_extensions==4.12.2
tzdata==2024.1
urllib3==2.2.2
Werkzeug==3.0.4
wrapt==1.16.0

### Reproducible example

```Python
from deepface import DeepFace

dfs = DeepFace.find(
  img_path = "reference2.jpg",
  db_path = "images",
)
```

### Relevant Log Output

24-08-31 18:00:35 - 🔴 Exception while extracting faces from images/30733230622172002062022.jpg: Face could not be detected in images/30733230622172002062022.jpg. Please confirm that the picture is a face photo or consider to set enforce_detection param to False.
Finding representations:   7%|█████████| 1056/15466 [09:44<2:22:37,  1.68it/s]

### Expected Result

It to actually be fast. 

### What happened instead?

The speed is only about 2it/s. I have a i7 4770 non-k, and a GTX 1080. 

I have the tensorflow GPU version installed. Which is done by pip installing "tensorflow[and-cuda]".

Running the tensorflow command to find if there are any GPU's returns my main GPU.

This does mean that my GPU is detected, it is the case that DeepFace does not actually use it?

**Comment by serengil** [comment]
deepface does nothing with gpu. it depends on tensorflow. if you can use your gpu with your tf installation, then it will be used in deepface, too.

**Comment by rtj563etn36h4h** [comment]
> deepface does nothing with gpu. it depends on tensorflow. if you can use your gpu with your tf installation, then it will be used in deepface, too.

I've read this same issue about 100 times in this repo, yet you have the same answer everytime. Have you considered actually looking if there might be a problem on your end?

**Comment by serengil** [comment]
No, i am not planning to focus on this at all. Because, I answered this question 100 times, you are still raising same issue.

**Comment by rtj563etn36h4h** [comment]
> No, i am not planning to focus on this at all. Because, I answered this question 100 times, you are still raising same issue.

Okay, then just for reference. Does an average of 2it/s sound like it's using the GPU to you? Around the level of a GTX 1080? 

Because I can't tell from the speed which is actually being used, since I'm new to this.

**Comment by serengil** [comment]
you can monitor the gpu activity with `watch nvidia-smi` command.

**Comment by rtj563etn36h4h** [comment]
> you can monitor the gpu activity with `watch nvidia-smi` command.

![image](https://github.com/user-attachments/assets/2f12fe2d-6b74-4b4e-8289-4671bdf4e168)

This is my usage. I see that the dedicated memory 7/8GB of it is being used. For all the other stats, it looks like they're at 0. I view these through the Windows task manager. Stuff like 3D and the like is at 0%. 

Even so, when I tell you my GPU and the speed of the task, does that seem nominal? The speed?

**Comment by serengil** [comment]
yes, gpu memory allocated and 9% is working. so, you are using your gpu.

**Comment by rtj563etn36h4h** [comment]
> yes, gpu memory allocated and 9% is working. so, you are using your gpu.

Okay and for my final question again, does the speed seem nominal?
